libbeat/common/docker/watcher.go

```diff
@@ -291,8 +293,15 @@ func (w *watcher) listContainers(options types.ContainerListOptions) ([]*Contain
 	var result []*Container
 	for _, c := range containers {
 		var ipaddresses []string
+		var gateways []string
 		for _, net := range c.NetworkSettings.Networks {
-			ipaddresses = append(ipaddresses, net.IPAddress)
+			if net.IPAddress != "" {
+				ipaddresses = append(ipaddresses, net.IPAddress)
+			}
+
+			if net.Gateway != "" {
+				gateways = append(gateways, net.Gateway)
+			}
```

Comment:
Here, when requesting information from the daemon we could also get the Hostname, at least if we use inspect, not sure with list.

Reply:
A container running in host network gets the hostname of the host.